relayer/src/listeners/advisory.rs

```diff
@@ -0,0 +1,161 @@
+use std::{
+    collections::HashMap,
+    sync::{
+        atomic::{AtomicBool, Ordering},
+        Arc,
+    },
+};
+
+use aleph_client::{
+    contract::{
+        event::{translate_events, BlockDetails, ContractEvent},
+        ContractInstance,
+    },
+    contract_transcode::Value,
+    AlephConfig,
+};
+use futures::StreamExt;
+use log::{debug, info};
+use subxt::{events::Events, utils::H256};
+
+use super::AzeroListenerError;
+use crate::{
+    config::Config,
+    connections::azero::AzeroWsConnection,
+    contracts::{decode_bool_field, AdvisoryInstance, AzeroContractError},
+};
+
+pub struct AdvisoryListener;
+
+impl AdvisoryListener {
```

Comment:
My intuition would be that a better approach is to just query the state of the contract every 5 seconds or so.
Response time would be similar, but the code would be much simpler + less load on the node

Reply:
Alternatively, we could take advantage of the fact that we are already fetching events from all blocks, but not sure how complex sharing that with advisory listener would be

Reply:
yeah, good idea, I will think of sharing this via a channel or something similar